### PR TITLE
Bug 2077506: backport fixes for OVS HW offload & dependencies

### PR DESCRIPTION
**Pull request by EmilienM**
- openstack: move metadata fetch under daemon writer and reduce runs
- virtual/openstack: configure VFIO with NIOMMU by default
- openstack: include support to metadata service
- Vendor hashicorp/go-retryablehttp deps
- Improve the virtual plugin support
- Fix the validation webhook for virtual environments
- Don't run the networkManager udev rule on virtual environments
- [downstream-only] add Red Hat Virtio into supported-nic ConfigMap

### CODE REVIEW

**Review by MaysaMacedo — COMMENTED** [review]
Just a few of questions:

Should we make sure that certain commit orders are preserved? For example:
`Vendor hashicorp/go-retryablehttp deps` to come before the `openstack: include support to metadata service`

File: manifests/4.10/supported-nic-ids_v1_configmap.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ data:
   Qlogic_qede_QL41000: 1077 8070 8090
   Qlogic_qede_QL45000_25G: 1077 1656 1664
   Qlogic_qede_QL45000_50G: 1077 1654 1664
+  Red_Hat_Virtio_network_device: "1af4 1000 1000"
```
[the remark this answers is not in the document]
Comment (EmilienM) [reply manifests/4.10/supported-nic-ids_v1_configmap.yaml]: done

**Comment by EmilienM** [comment]
> Just a few of questions:
> 
> Should we make sure that certain commit orders are preserved? For example: `Vendor hashicorp/go-retryablehttp deps` to come before the `openstack: include support to metadata service`

done!

**Comment by eurijon** [comment]
@zhaozhanqi @mffiedler hey, as we don't have 'cherry-pick-approval' rights in this repo, would you (or someone from your team) be able to add the label in this PR?
The BZ in 4.11 (https://bugzilla.redhat.com/show_bug.cgi?id=2036948) has been successfully verified.
Thanks!

**Comment by mffiedler** [comment]
/label cherry-pick-approved

@eurijon Done.

**Pull request merged**